### PR TITLE
Add CONTRIBUTING file

### PR DESCRIPTION
**Pull request by dmathieu**
This adds a CONTRIBUTING file, so new contributors better know what to expect.
I based this on the [otel-go contributing file](https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md), with some adjustements.
Everything is open for comments obviously.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (christos68k) [inline CONTRIBUTING.md:17]: For consistency with the instructions that follow:

```suggestion
git clone https://github.com/open-telemetry/opentelemetry-ebpf-profiler
```

File: CONTRIBUTING.md
Comment (christos68k) [inline CONTRIBUTING.md:40]:
```suggestion
This will put the project in `opentelemetry-ebpf-profiler` in
```

**Review by christos68k — COMMENTED** [review]
Did a quick pass

File: CONTRIBUTING.md
Comment (florianl) [inline CONTRIBUTING.md:6]: There are many SIG meetings and over time there will be even more. Maybe [calendar-profiling](https://groups.google.com/a/opentelemetry.io/g/calendar-profiling) or the slack channel [#otel-profiles](https://cloud-native.slack.com/archives/C03J794L0BV) could be mentioned for easier access.

File: CONTRIBUTING.md
Comment (christos68k) [inline CONTRIBUTING.md:87]: AFAIK we need two approvals for all PRs, so this should probably be reworded. 

CC: @athre0z

File: CONTRIBUTING.md
Comment (florianl) [inline CONTRIBUTING.md:24]: There are multiple Make targets. Might be a personal view, but `make codespell` is not important to get started. To start with a project, targets on how to build and how to test are the most important ones, I think.
Otherwise, the question arises, why `codespell` should be mentioned here but not `lint`.
```suggestion
```

File: CONTRIBUTING.md
Comment (christos68k) [inline CONTRIBUTING.md:114]: Same here, we require two approvals for all PRs.

File: CONTRIBUTING.md
Comment (florianl) [inline CONTRIBUTING.md:140]: Things change and so do memberships of such groups. To keep it more generic, https://github.com/orgs/open-telemetry/teams/ebpf-profiler-approvers and https://github.com/orgs/open-telemetry/teams/ebpf-profiler-maintainers should be mentioned here instead of personal accounts.
```suggestion
@open-telemetry/ebpf-profiler-approvers and @open-telemetry/ebpf-profiler-maintainers 
```

Comment (dmathieu) [reply CONTRIBUTING.md:137]: While I'm happy to mention the groups and not the folks here, every other SIG does it like this.
And updating this list is part of the maintainer onboarding tasks.

https://github.com/open-telemetry/community/pull/1924/files#diff-2babc18ce7c6fdc68c4161df08fb505a258f0bce4dda17f081d7628b0afa04c7R64

File: CONTRIBUTING.md
Comment (rockdaboot) [reply CONTRIBUTING.md:91]: Our experience is that opinions about what is trivial or not are different. To avoid friction and potential conflicts, we decided to always require 2 approvals. Especially for trivial changes, getting two approvals should not be an issue. We can also discuss this decision any time on Slack or in a profiling SIG meeting.

File: CONTRIBUTING.md
Comment (athre0z) [inline CONTRIBUTING.md:141]: Training `-` here: no sure if intentional?

Comment (athre0z) [inline CONTRIBUTING.md:141]: Trailing `-` here: no sure if intentional?

```suggestion
- [Dmitry Filimonov](https://github.com/petethepig), Pyroscope/Grafana
```

File: CONTRIBUTING.md
Comment (athre0z) [inline CONTRIBUTING.md:143]: Space before comma

```suggestion
- [Joel Höner](https://github.com/athre0z), Elastic
```

File: CONTRIBUTING.md
Comment (athre0z) [inline CONTRIBUTING.md:17]: Wondering whether we should move this down, into a dedicated new section, e.g. "Community". Starting the document with this may give the impression that one has to take part in these meetings in order to contribute?

File: CONTRIBUTING.md
Comment (florianl) [reply CONTRIBUTING.md:144]: Fair point - I wasn't aware of this additional manual step in this process.

File: CONTRIBUTING.md
Comment (dmathieu) [reply CONTRIBUTING.md]: I have removed trivial changes.

**Review by florianl — APPROVED** [review]

**Review by rockdaboot — APPROVED** [review]

**Review by athre0z — APPROVED** [review]
LGTM, thanks for tackling this! :)

**Pull request merged**
